cognite/neat/rules/models/_rules/dms_schema.py

```diff
@@ -67,6 +85,171 @@ def from_model_id(cls, client: CogniteClient, data_model_id: dm.DataModelIdentif
+        data = cls._read_directory(Path(directory))
+        return cls.load(data)
+
+    @classmethod
+    def _read_directory(cls, directory: Path) -> dict[str, list[Any]]:
+        data: dict[str, Any] = {}
+        for yaml_file in directory.rglob("*.yaml"):
+            if "." in yaml_file.stem:
+                resource_type = yaml_file.stem.rsplit(".", 1)[-1]
+                if attr_name := cls._FIELD_NAME_BY_RESOURCE_TYPE.get(resource_type):
+                    data.setdefault(attr_name, [])
+                    loaded = yaml.safe_load(yaml_file.read_text())
+                    if isinstance(loaded, list):
+                        data[attr_name].extend(loaded)
+                    else:
+                        data[attr_name].append(loaded)
+        return data
+
+    def to_directory(self, directory: str | Path, exclude: set[str] | None = None) -> None:
+        """Save the schema to a directory as YAML files. This is compatible with the Cognite-Toolkit convention.
+
+        Args:
+            directory (str | Path): The directory to save the schema to.
+            exclude (set[str]): A set of attributes to exclude from the output.
+        """
+        path_dir = Path(directory)
+        exclude_set = exclude or set()
+        data_models = path_dir / "data_models"
+        data_models.mkdir(parents=True, exist_ok=True)
+        if "spaces" not in exclude_set:
+            for space in self.spaces:
+                (data_models / f"{space.space}.space.yaml").write_text(space.dump_yaml())
+        if "data_models" not in exclude_set:
+            for model in self.data_models:
+                (data_models / f"{model.external_id}.datamodel.yaml").write_text(model.dump_yaml())
+        if "views" not in exclude_set and self.views:
+            view_dir = data_models / "views"
+            view_dir.mkdir(parents=True, exist_ok=True)
+            for view in self.views:
+                (view_dir / f"{view.external_id}.view.yaml").write_text(view.dump_yaml())
+        if "containers" not in exclude_set and self.containers:
+            container_dir = data_models / "containers"
+            container_dir.mkdir(parents=True, exist_ok=True)
+            for container in self.containers:
+                (container_dir / f"{container.external_id}.container.yaml").write_text(container.dump_yaml())
+        if "node_types" not in exclude_set and self.node_types:
+            node_dir = data_models / "nodes"
+            node_dir.mkdir(parents=True, exist_ok=True)
+            for node in self.node_types:
+                (node_dir / f"{node.external_id}.node.yaml").write_text(node.dump_yaml())
+
+    @classmethod
+    def from_zip(cls, zip_file: str | Path) -> Self:
+        """Load a schema from a ZIP file containing YAML files.
+
+        The ZIP file is expected to follow the Cognite-Toolkit convention
+        where each file is named as `resource_type.resource_name.yaml`.
+        """
+        data = cls._read_zip(Path(zip_file))
```

Comment:
I would be ok if there is additional logic following cls._read_zip, but as it is, why not moving code from _read_zip here?